src/System.Text.JsonLab/System/Text/Json/JsonWriterUtf16.cs

```diff
@@ -543,11 +543,34 @@ public void WriteValue(Guid value)
         /// </summary>
         public void WriteNull()
         {
-            //TODO: Optimize, just like WriteValue(long value)
-            WriteItemSeperatorUtf16();
-            _firstItem = false;
-            WriteSpacingUtf16();
+            int charsNeeded = (_firstItem ? 0 : 1) + (_prettyPrint ? 2 + (_indent + 1) * 2 : 0);
+            int bytesNeeded = charsNeeded * 2;
+            Span<char> charBuffer = MemoryMarshal.Cast<byte, char>(EnsureBuffer(bytesNeeded));
+            int idx = 0;
+            if (_firstItem)
+            {
+                _firstItem = false;
+            }
+            else 
+            {
+                charBuffer[idx++] = (char)JsonConstants.ListSeperator;
+            }
 
+            if (_prettyPrint) 
+            {
+                int indent = _indent;
+                charBuffer[idx++] = (char)JsonConstants.CarriageReturn;
+                charBuffer[idx++] = (char)JsonConstants.LineFeed;
+
+                while (indent-- >= 0)
+                {
+                    charBuffer[idx++] = (char)JsonConstants.Space;
+                    charBuffer[idx++] = (char)JsonConstants.Space;
+                }
+            }
+
+            _bufferWriter.Advance(bytesNeeded);
+            Debug.Assert(idx == charsNeeded);
             WriteJsonValue(JsonConstants.NullValue);
```

Comment:
I think we should do something similar here as Utf8 (and use copyto).

Given the value we are writing is a constant, it doesn't make sense to keep calling TryEncode on it. We should create a UTF-16 encoded NullValue as a byte array and write that to the output buffer
https://github.com/dotnet/corefxlab/blob/3b9952e99cbd50c9018e7976b31e08820b821a57/src/System.Text.Primitives/System/Text/SymbolTable/SymbolTable_utf16.cs#L54-L64

```C#
public static readonly byte[] NullValueUtf16 = { (byte)'n', 0, (byte)'u', 0, (byte)'l', 0, (byte)'l', 0 };
```